### PR TITLE
`Development`: Update server dependencies

### PR DESCRIPTION
**Pull request by krusche**
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] All relevant checks pass.

### CODE REVIEW

**Comment by basak-akan** [comment]
Testing session tasks

New Artemis User
- [ ] Password generated
- [ ] Participated in programming exercise
- [ ] Participated in text exercise
- [ ] Programming exercise results visible
- [ ] Text exercise results visible

Artemis User
- [ ] Participate in programming exercise
- [ ] Programming exercise results visible
- [ ] Participate text exercise
- [ ] Text exercise results visible
- [ ] Participate in modeling exercise
- [ ] Modeling exercise results visible
- [ ] Participate in drag and drop exercise
- [ ] Drag-and-drop exercise results are visible

**Review by max-bergmann — APPROVED** [review]
Tested in Testing session - works for me as expected

**Review by rstief — APPROVED** [review]
Tested in Testing Session.

**Review by vinceclifford — APPROVED** [review]
Works as expected. Tested in the testing session on Nov 2nd

**Review by lennart-keller — APPROVED** [review]
Tested on ts11 with test user 4. Works as expected

**Review by florian-glombik — APPROVED** [review]
Tested during testing session, the password was generated as expected, the results for the different exercise types were displayed

File: src/main/java/de/tum/in/www1/artemis/config/PublicResourcesConfiguration.java [hunk 1]
```diff
@@ -23,7 +23,7 @@
 @Configuration
 public class PublicResourcesConfiguration implements WebMvcConfigurer {
 
-    @Autowired
+    @Autowired // ok
```
Comment (laurenzfb) [inline src/main/java/de/tum/in/www1/artemis/config/PublicResourcesConfiguration.java:26]: What's the reason for the //ok comments?

File: src/main/java/de/tum/in/www1/artemis/service/listeners/ResultListener.java [hunk 2]
```diff
@@ -27,7 +27,7 @@ public ResultListener() {
         // Empty constructor for Spring
     }
 
-    @Autowired
+    @Autowired // ok
```
Comment (laurenzfb) [inline src/main/java/de/tum/in/www1/artemis/service/listeners/ResultListener.java:30]: .

**Review by laurenzfb — APPROVED** [review]
Code changes seem reasonable to me. Left a comment

**Pull request merged**